swarm/storage/mru/doc.go

```diff
@@ -1,52 +1,60 @@
 // which are stored as little-endian uint64 values in the database (for a
-// total of 16 bytes). It also contains the unique identifier.
+// total of 16 bytes). It also contains the owner's address (ownerAddr)
 // This MRU info is stored in a separate content-addressed chunk
 // (call it the metadata chunk), with the following layout:
 //
-// (startblock|frequency|address|identifier)
+// (00|length|startTime|frequency|name|ownerAddr)
 //
 // (The two first zero-value bytes are used for disambiguation by the chunk validator,
 // and update chunk will always have a value > 0 there.)
 //
+// Each metadata chunk is identified by its rootAddr, calculated as follows:
+// metaHash=H(len(metadata), startTime, frequency,name)
+// rootAddr = H(metaHash, ownerAddr).
+// where H is the SHA3 hash function
+// This scheme effectively locks the root chunk so that only the owner of the private key
+// that ownerAddr was derived from can sign updates.
+//
 // The root entry tells the requester from when the mutable resource was
-// first added (block number) and in which block number to look for the
+// first added (Unix time in seconds) and in which moments to look for the
 // actual updates. Thus, a resource update for identifier "føø.bar"
-// starting at block 4200 with frequency 42 will have updates on block 4242,
-// 4284, 4326 and so on.
+// starting at unix time 1528800000 with frequency 300 (every 5 mins) will have updates on 1528800300,
+// 1528800600, 1528800900 and so on.
 //
 // Actual data updates are also made in the form of swarm chunks. The keys
 // of the updates are the hash of a concatenation of properties as follows:
 //
-// sha256(period|version|address|namehash)
-//
-// The period is (currentblock - startblock) / frequency
+// updateAddr = H(period, version, rootAddr)
+// where H is the SHA3 hash function
+// The period is (currentTime - startTime) / frequency
 //
-// Using our previous example, this means that a period 3 will have 4326 as
-// the block number.
+// Using our previous example, this means that a period 3 will happen when the
+// clock hits 1528800900
 //
-// If more than one update is made to the same block number, incremental
+// If more than one update is made in the same period, incremental
 // version numbers are used successively.
 //
-// A lookup agent need only know the identifier name in order to get the versions
+// A lookup agent need only know the rootAddr in order to get the versions
 //
-// the resourcedata is:
-// headerlength|period|version|identifier|data
+// the resource update data is:
+// resourcedata = headerlength|period|version|metaHash|rootAddr|data
 //
 // the full update data that goes in the chunk payload is:
 // resourcedata|sign(resourcedata)
 //
-// headerlength is a 16 bit value containing the byte length of period|version|name
+// headerlength is a 16 bit value containing the byte length of period|version|rootAddr|metaHash
```

Comment:
with rootAddr being a hash length, headerlength is not needed, was it not meant to delineate signature from data?
best structure is

`period|version|metaHash|signature|data`

Reply:
Including a header length field is always a good practice when encoding binary data, for extensibility and error checking.

The header of an `updateRequest` object does not contain the data.

Let's take the encoding discussion to #704, where it can be clearly seen in the refactored code.